openbb_platform/providers/fred/openbb_fred/models/bond_indices.py

```diff
@@ -0,0 +1,590 @@
+    )
+
+
+class FredBondIndicesFetcher(
+    Fetcher[
+        FredBondIndicesQueryParams,
+        List[FredBondIndicesData],
+    ]
+):
+    """FRED Bond Indices Fetcher."""
+
+    @staticmethod
+    def transform_query(params: Dict[str, Any]) -> FredBondIndicesQueryParams:
+        """Transform query."""
+        values = params.copy()
+        new_index = []
+        messages = []
+        if "index" not in values:
+            values["index"] = "yield_curve"
+        if "category" not in values:
+            values["category"] = "us"
+        if "index_type" not in values:
+            values["index_type"] = "yield"
+        is_yield_curve = False
+        if "yield_curve" in values["index"]:
+            values["category"] = "us"
+            values["index"] = "yield_curve"
+            new_index.append("yield_curve")
+            is_yield_curve = True
+            if (
+                isinstance(values["index"], list)
+                and len(values["index"] > 1)
+                or isinstance(values["index"], str)
+                and "," in values["index"]
+            ):
+                message = "Multiple indices not allowed for: 'yield_curve'."
+                messages.append(message)
+        if is_yield_curve is False:
+            indices = (
+                values["index"]
+                if isinstance(values["index"], list)
+                else values["index"].split(",")
+            )
+            for index in indices:
+                if values["category"] == "us":
+                    if index not in BAML_CATEGORIES.get("us"):
+                        message = (
+                            "Invalid index for category: 'us'."
+                            + f" Must be one of {','.join(BAML_CATEGORIES.get('us'))}."
+                        )
+                        messages.append(message)
+                    else:
+                        new_index.append(index)
+                if values["category"] == "high_yield":
+                    if index not in ("us", "europe", "emerging"):
+                        message = (
+                            "Invalid index for category: 'high_yield'."
+                            + f" Must be one of {','.join(BAML_CATEGORIES.get('high_yield', ''))}."
+                        )
+                        messages.append(message)
+                    else:
+                        new_index.append(index)
+                if values["category"] == "emerging_markets":
+                    if index not in BAML_CATEGORIES.get("emerging_markets"):
+                        message = (
+                            "Invalid index for category: 'emerging_markets'."
+                            + f" Must be one of {','.join(BAML_CATEGORIES.get('emerging_markets', ''))}."
+                        )
+                        messages.append(message)
+                    else:
+                        new_index.append(index)
+        if not new_index:
+            raise ValueError(
```

Comment:
If these error messages are for the user it should be raising `OpenBBError` instead.

`ValueError` (or any python base exception) will display `Unexpected Error` in the UI